src/Functions/array/arrayPrAUC.cpp

```diff
@@ -0,0 +1,246 @@
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
```

Comment:
The code in this file seems to have 80% overlap with arrayAUC.cpp. That's too much copypasta. Please make a single .cpp file from both files, the different parts can be distinguished using templatization.

Reply:
The "scaling" option used to calculate the `arrayRocAUC` in parallel won't work for the `arrayPrAUC`. If a way to calculate the PR AUC on a part of the curve was added, the implementation and arguments end up differing quite a bit. 

I opened a issue discussing the parallelization of the PR AUC here https://github.com/ClickHouse/ClickHouse/issues/72942. I would love to hear your thoughts regarding a way to do so and if we could find a common ground between them.